tests/src/system/packages/MessagingFeedTests.scala

```diff
@@ -0,0 +1,190 @@
+
+import java.io._
+import java.util.Properties
+
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+import scala.language.postfixOps
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.junit.runner.RunWith
+import org.scalatest.BeforeAndAfterAll
+import org.scalatest.FlatSpec
+import org.scalatest.Matchers
+import org.scalatest.junit.JUnitRunner
+
+import whisk.connector.kafka.KafkaProducerConnector
+import whisk.core.WhiskConfig
+import whisk.core.connector.Message
+import whisk.utils.ExecutionContextFactory
+import common.JsHelpers
+import common.TestHelpers
+import common.TestUtils
+import common.Wsk
+import common.WskActorSystem
+import common.WskProps
+import common.WskTestHelpers
+import spray.json.DefaultJsonProtocol.StringJsonFormat
+import spray.json.JsArray
+import spray.json.JsString
+import spray.json.pimpAny
+
+@RunWith(classOf[JUnitRunner])
+class MessagingFeedTests 
+    extends FlatSpec
+    with Matchers
+    with WskActorSystem
+    with BeforeAndAfterAll
+    with TestHelpers
+    with WskTestHelpers
+    with JsHelpers {
+    
+    val groupid = "kafkatest"
+    val topic = "test"
+    val kafka_topic = "Dinosaurs"
+    val sessionTimeout = 10 seconds
+
+    implicit val wskprops = WskProps()
+    val wsk = new Wsk()
+
+    val messagingPackage = "/whisk.system/messaging"
+    val kafkaFeed = "kafkaFeed"
+    val messageHubFeed = "messageHubFeed"
+
+    behavior of "Kafka connector"
+
+    it should "fire a trigger when a message is posted to Kafka" in withAssetCleaner(wskprops) {
+        val openwhisk_home = System.getenv("OPENWHISK_HOME")
+        val file = new File(openwhisk_home + "/whisk.properties")
+        val config = new WhiskConfig(Map("kafka.host" -> null, "kafka.host.port" -> null),
+            Set(), file)
+        // Check if the Kafka host and port are valid in the property file.
+        assert(config.isValid)
+        implicit val ec = ExecutionContextFactory.makeCachedThreadPoolExecutionContext()
+        val producer = new KafkaProducerConnector(config.kafkaHost, ec)
+        (wp, assetHelper) =>
+            val triggerName = s"dummyKafkaTrigger-${System.currentTimeMillis}"
+
+            // Make sure the package of the Kafka feed exists.
+            val packageGetResult = wsk.pkg.get(messagingPackage)
+            packageGetResult.stdout should include("ok")
+
+            // Create a trigger with the Kafka feed.
+            var kafka_brokers = JsArray(JsString(config.kafkaHost))
+            val feedCreationResult = assetHelper.withCleaner(wsk.trigger, triggerName) {
+                (trigger, name) =>
+                    trigger.create(name, feed = Some(s"$messagingPackage/$kafkaFeed"), parameters = Map(
+                        "brokers" -> kafka_brokers,
+                        "topic" -> kafka_topic.toJson))
+            }
+            feedCreationResult.stdout should include("ok")
```

Comment:
While I'm sure you have seen this check done in other tests, it is not very robust. For example, if stdout contains "The result was **not** ok" then this check will pass because it only looks for the string "ok" somewhere in stdout.

I really like using `withActivation()` followed by `activation.response.success shouldBe true`. `withActivation` does the polling for you, as well as results in an object that is already parsed and ready for thorough inspection.

You can find a good example here: https://github.com/openwhisk/openwhisk/blob/master/tests/src/system/basic/Swift3WhiskObjectTests.scala#L56